src/components/smart_objects/include/smart_objects/schema_item.h

```diff
@@ -91,13 +101,27 @@ class ISchemaItem {
    **/
   virtual bool hasDefaultValue(SmartObject& Object);
 
+  /**
+   * @brief Filter invalid enum values
+   *
+   * @param Object Object to check for invalid enum values
+   * @param MessageVersion the version of the schema to use for validation
+   * @param report__ object for reporting enums which were removed during the
+   * process
+   *
+   * @return true if the value being checked should be filtered, false otherwise
+   **/
+  virtual bool filterInvalidEnums(SmartObject& Object,
```

Comment:
I see check of return value only in unit tests. Do we need it?

Reply:
Yes, it is used recursively within the function. If a particular element returns true, that means that it should be removed from the containing structure. This return value is not used at the root level because that would mean that the full structure would be destroyed.